src/gui/properties/propertieswidget.cpp

```diff
@@ -440,13 +440,15 @@ void PropertiesWidget::loadDynamicData()
                 .arg(QString::number(m_torrent->leechsCount())
                     , QString::number(m_torrent->totalLeechersCount())));
 
+            const int dlDuration = m_torrent->activeTime() - m_torrent->finishedTime();
+            const QString dlAvg = Utils::Misc::friendlyUnit((m_torrent->totalDownload() / ((dlDuration == 0) ? 1 : dlDuration)), true);
```

Comment:
But in case of `(dlDuration == 0)` the printed avg speed will be still wrong. Why not print it as "Uknown" aka
```c++
((dlDuration <= 0) ? -1 : dlDuration)
```

???

Reply:
Or from another viewpoint, I think my proposed form is actually closer to what we intend to do here: avoiding `0` explicitly and also preserve the side-effect of printing "Unknown" when duration value is <0.

Reply:
The you end up printing an "avg speed" based on total bytes, which doesn't make sense and will result in big numbers for big torrents.

Reply:
When time is 0 the result should be constantly 0.

Reply:
I should revise my viewpoint, the result of any number div by zero is undefined, thus printing "Unknown" in this case makes sense.
PR is updated to reflect this.